### ISSUE TITLE
not able to find way to add new key-value pair and how to initialize i18n with value from mongoDB

### ISSUE THREAD

**Issue by premnkl** [lamualfa/i18next-node-mongodb-backend-next#21]
Dear Team,

**Need help for below points. I am not getting references for below things:**
1. How do we initialize the mongoDB with language data
2. How to upsert existing key value pair
3. How i18n in react initializes with the mongoDB data as resource

I followed the readMe and initialized the i18next using nodeJS as below, which created a new table in mongoDB:

i18next.use(Backend).init({
  // Backend Options
  backend: {
    host: 'localhost',
    port: 27017,
    dbName: 'dy-mde', // Required field
  },
})_

![image](https://user-images.githubusercontent.com/108449384/190596012-9fb0a158-f844-4997-b178-1215e53b5eb1.png)

Regards,
Prem

**Comment by premnkl** [comment]
@Unknown051 @lamualfa @mobaabid please for your comments

**Comment by lamualfa** [comment]
> How do we initialize the mongoDB with language data

The initializing process should be done in i18next layer. A backend layer like this library can't do it directly. See https://www.i18next.com/how-to/add-or-load-translations for further information.

<hr>

> How to upsert existing key value pair

You can update the locale data directly via i18next. This Backend plugin with update it automatically.

<hr>

> How i18n in react initializes with the mongoDB data as resource

I am not sure I fully understand this question.

If you mean to use i18next in the React.js application, you can follow this guide https://react.i18next.com/getting-started.

But if you mean the mechanism that i18next use to interact with their backend, the backend will implement some of the abstraction methods that i18next provided before. For further information, you can follow this guide about creating your own plugin https://www.i18next.com/misc/creating-own-plugins#backend.

**Comment by lamualfa** [comment]
> I followed the readMe and initialized the i18next using nodeJS as below, which created a new table in mongoDB:

Yeah, it's already correct. The collection is still empty because you don't initialize any data using i18next.

**Comment by premnkl** [comment]
@lamualfa thanks for your comments. **Please for your inputs to below cases what we have**

we have 2 cases to implement
1. To use mongoDB to store all the translation values and to fetch the value from mongoDB as a source for initializing i18next

**similar to below sample code**
_i18next.use(Backend).use(middleware.LanguageDetector).init({
		debug: true,
		load: ['ar','en'],
		fallbackLng: 'ar',
		backend: { 
			uri: 'mongodb://localhost/test',
			collection: 'translations'
		},_

2. Providing user an option to add new translation text on runtime. Once we receive the input from user, we need to update the existing key-value pair in mongoDB with the new value received. Once mongoDB is updated with user input. We need to reload the resource for i18n

we believe this can be implemented with the help of '_i18next-node-mongodb-backend-next_'


regards,
prem

**Comment by lamualfa** [comment]
> To use mongoDB to store all the translation values and to fetch the value from mongoDB as a source for initializing i18next

You must insert and read the translation data via i18next, not directly into the database. The Backend layer, like this library, doesn't provide any of those features.

<hr>

> Providing user an option to add new translation text on runtime. Once we receive the input from user, we need to update the existing key-value pair in mongoDB with the new value received. Once mongoDB is updated with user input. We need to reload the resource for i18n

Use `t` method to read the translation data. Use `addResource` method to add or update the translation data. Use `reloadResources` to synchronize data between i18next and the database. Those methods was provided by i18next.

**References**

- https://www.i18next.com/overview/api?q=update#t
- https://www.i18next.com/overview/api#addresource
- https://www.i18next.com/overview/api#reloadresources

**Comment by premnkl** [comment]
@lamualfa thanks for quick response

sorry for more questions, I am trying to understand how I can use i18next-node-mongodb-backend-next

from below point in the readMe
_This is a i18next backend to be used Node JS. It will load resources from a MongoDB database with official node mongodb_ 

I understood this package can be used to load data from mongoDB for i18next initialization. Am I right ?

**Comment by lamualfa** [comment]
> I understood this package can be used to load data from mongoDB for i18next initialization. Am I right ?

Yeah, of course. This library has been officially approved by i18next team.

**Comment by premnkl** [comment]
@lamualfa I am looking for sample to understand how to do the same

can u help me with that ?

thanks,
prem

**Comment by lamualfa** [comment]
> @lamualfa I am looking for sample to understand how to do the same
> 
> can u help me with that ?
> 
> thanks,
> prem

Try looking these examples https://github.com/lamualfa/i18next-node-mongodb-backend-next/tree/v0.x-examples.

**Comment by premnkl** [comment]
@lamualfa I am going through the examples

will comeback, If I need more info

might trouble you again with questions, sorry for that :)


Regards,
prem
